cmake/podioMacros.cmake

```
@@ -151,3 +151,40 @@ function(PODIO_GENERATE_DATAMODEL datamodel YAML_FILE RETURN_HEADERS RETURN_SOUR
     )
 
 endfunction()
+
+#---------------------------------------------------------------------------------------------------
+function(PODIO_CHECK_CPP_FS FS_LIBS)
+  SET(${have_filesystem} FALSE)
+  MESSAGE(VERBOSE "Checking for filesystem support of compiler")
+  # GNU implementation prior to 9.1 requires linking with -lstdc++fs and LLVM
+  # implementation prior to LLVM 9.0 requires linking with -lc++fs
+  # After that it should be built-in
+  FOREACH(FS_LIB_NAME "" stdc++fs c++fs)
+    MESSAGE(VERBOSE "Linking against ${FS_LIB_NAME}")
+    try_compile(have_filesystem ${CMAKE_BINARY_DIR}/try ${PROJECT_SOURCE_DIR}/cmake/try_filesystem.cpp
+      CXX_STANDARD ${CMAKE_CXX_STANDARD}
+      CXX_EXTENSIONS False
+      OUTPUT_VARIABLE HAVE_FS_OUTPUT
+      LINK_LIBRARIES ${FS_LIB_NAME}
+      )
+    MESSAGE(DEBUG "-----> " ${HAVE_FS_OUTPUT})
+    IF(have_filesystem)
+      MESSAGE(VERBOSE "Compiler supports filesystem when linking against ${FS_LIB_NAME}")
+      SET(${FS_LIBS} ${FS_LIB_NAME} PARENT_SCOPE)
+      BREAK()
```

```suggestion
      RETURN()
```
As this is a function, we can use RETURN(). It makes no real difference, but we could drop the `IF(NOT have_filesystem)`